packages/shared/src/hooks/useAuthErrors.ts

```diff
@@ -1,10 +1,15 @@
-import useWindowEvents from './useWindowEvents';
-import { AuthEvent, getKratosError, ErrorEvent } from '../lib/kratos';
+import { AuthEvent, getKratosError } from '../lib/kratos';
 import { useToastNotification } from './useToastNotification';
+import { useEventListener } from './useEventListener';
 
 export function useAuthErrors(): void {
   const { displayToast } = useToastNotification();
-  useWindowEvents<ErrorEvent>('message', AuthEvent.Error, async (e) => {
+
+  useEventListener(globalThis, 'message', async (e) => {
+    if (e.data?.eventKey !== AuthEvent.Error) {
+      return;
+    }
+
     if (!e.data?.id) {
```

Comment:
Since the return value is the same, these conditions can be merged, e.g. 
```suggestion
    if (e.data?.eventKey !== AuthEvent.Error || !e.data?.id) {
```

Reply:
Would it not be less readable then?

Reply:
I don't have a strong preference. To me the readability is the same. Having 2 conditions below each other when it can be defined with OR operator, I always prefer the OR operator.

If readability is a concern, you can split the condition into two lines, e.g.:
```suggestion
    if (e.data?.eventKey !== AuthEvent.Error ||
        !e.data?.id)
      return;
    }
```

Reply:
Also, do not feel obliged. This is not blocking the merge from my point of view, just to be clear :)

Reply:
Above would not be allowed by our linter but basically not about code readability, its about logic readability. 

Multiple `||` and `&&` etc. quickly become hard to parse logically when looking the code. Throw in a negation `!` and you are looking at something that becomes a monster really fast. 

Early return pattern like above solves it in 99% of the time while keeping conditions nice and lean. Would love to talk about it more at some other time 🙏